src/main/java/com/ldtteam/structurize/placement/handlers/placement/PlacementHandlers.java

```diff
@@ -63,6 +99,64 @@ private PlacementHandlers()
          */
     }
 
+    public static class FluidSubstitutionPlacementHandler implements IPlacementHandler
+    {
+        @Override
+        public boolean canHandle(@NotNull World world, @NotNull BlockPos pos, @NotNull BlockState blockState)
+        {
+            return blockState.getBlock() instanceof BlockFluidSubstitution;
+        }
+
+        @Override
+        public List<ItemStack> getRequiredItems(@NotNull World world, @NotNull BlockPos pos, @NotNull BlockState blockState, @Nullable CompoundNBT tileEntityData, boolean complete)
+        {
+            return new ArrayList<>();
```

Comment:
not sure if this should return bucket with fluid

Reply:
oh, at least for lava it should